### ISSUE TITLE
Install cmake targets do not properly load their transitive dependencies.

### ISSUE THREAD

**Issue by johnb003** [curl/curl#2746]
### I did this

I am using a relocatable installation folder `${DIST}` for my cmake libs and openssl is installed there.
I can build referencing that installation just fine:

```
mkdir build && cd build
install_prefix=temp
DIST=`pwd`/../../../prebuilt/Darwin
cmake -DBUILD_SHARED_LIBS:BOOL=Off \
      -DCMAKE_POSITION_INDEPENDENT_CODE=On \
      -DCMAKE_INSTALL_PREFIX=${install_prefix} \
      -DCMAKE_PREFIX_PATH="${DIST}" \
      -DBUILD_TESTING=OFF \
      -DBUILD_CURL_EXE=OFF \
      -DCURL_STATICLIB=ON \
      -DENABLE_MANUAL=OFF \
      ..
make
make install
```

However, of the resulting installed config files,

```
> find temp/lib/cmake
temp/lib/cmake
temp/lib/cmake/curl
temp/lib/cmake/curl/curl-config-version.cmake
temp/lib/cmake/curl/curl-config.cmake
temp/lib/cmake/curl/libcurl-target-noconfig.cmake
temp/lib/cmake/curl/libcurl-target.cmake

```

...none appear to call find_dependency or find_package ultimately for any of the dependencies.

As a result, when my program attempts to link with a curl target, which was found with find_package config mode, it's incomplete.

I have to admit, I have had to work around the fact that my installation needs to be relocatable, and this target is also not relocatable because it's dumping the CURL_LIBRARIES to the export, which contains absolute paths. I have 2 ways I can fix this:

1. By changing the CMakeLists.txt to link against OpenSSL::SSL and OpenSSL::Crypto instead of OPENSSL_LIBRARIES (this results in these targets not being found -- due to the issue being reported)
2. By scrubbing the libcurl-target.cmake like so:
```
<   INTERFACE_LINK_LIBRARIES "lber;ldap;dl;/full/path/to/lib/libssl.dylib;/full/path/to/lib/libcrypto.dylib;/full/path/to/lib/libz.dylib"
---
>   INTERFACE_LINK_LIBRARIES "lber;ldap;dl;-lssl;-lcrypto;-lz"
```
Which will fail with `ld: library not found for -lssl` because, it didn't actually find it, and it's not actually in the system, and cmake won't add link directories.

The first option above is the better choice, as described here:
https://cmake.org/cmake/help/v3.12/manual/cmake-packages.7.html#creating-relocatable-packages

So, I guess the problem only really shows up if you first fix the problem of the importable target not being relocatable.

### I expected the following

I expected the config mode to actually find the libs it depends on not just add the link libraries for the target.

Canonically, this is done in the curl-config.cmake, and it would look like:
include(CMakeFindDependencyMacro)
find_dependency(OpenSSL)

Modern cmake would have also linked to OpenSSL::SSL and OpenSSL::Crypto in the first place, which also has the benefit of being relocatable.

### curl/libcurl version

my master branch is synced to 146178a9d

### operating system

Mac OSX High Sierra 10.13.5

**Comment by snikulov** [comment]
@johnb003 curl does not support exported targets, because it was written for CMake 2.6 initially. 
If you'll be able to invest some time into this - PR are welcoming.

**Comment by snikulov** [comment]
@johnb003 I was wrong. Looks like @jzakrzewski already added export abilities, but honestly, I never used it with curl project.

@jzakrzewski Could you please take a look on this issue?

**Comment by johnb003** [comment]
Here's a simple fix for the issue, if you're ok with how I fixed it.

**Comment by jzakrzewski** [comment]
@snikulov I may have some time today to check if @johnb003 's solution (minus imported targets which we cannot use with 2.8.12) does the job.

**Comment by snikulov** [comment]
@jzakrzewski @bagder @bradking perhaps it's time to increase `cmake_minimum_required ` for cmake?

**Comment by jzakrzewski** [comment]
I always think that getting a newer CMake version is easy enough not to be concerned about requiring "too new" version (one don't even have to install it - just unpack, in worst case build it; People that cannot do this are probably using autotools anyway). The 2.8.12 has been a compromise last time.

**Comment by bradking** [comment]
For reference, CMake 3.4 is the minimum required for FindOpenSSL to provide imported targets.  One could either require that or switch behavior based on availability of the targets.  The latter approach would allow users with older CMake versions to still build `curl` if that is all they care about.

**Comment by johnb003** [comment]
FWIW, my opinion is to bump your minimum cmake version. Like @jzakrzewski said, "getting a newer CMake version is enough".

More projects with higher versions adds pressure for everyone to have more recent versions, which means developers feel less resistance using modern features, and everyone wins. And CMake has gotten a LOT better over the recent years, and we should be pushing people to adopt the new style.

**Comment by snikulov** [comment]
closed with #2753
